blob/gcsblob/gcsblob.go

```diff
@@ -134,6 +148,12 @@ const Scheme = "gs"
 //
 //   - access_id: sets Options.GoogleAccessID
 //   - private_key_path: path to read for Options.PrivateKey
+//
+// Currently their use is limited to SignedURL. If left empty they will be guessed
+// which can result in inconsistencies and/or silently fail in environments such as Compute Engine.
```

Comment:
Please explain how they are guessed. IIUC, this will only work with a credentials file found on the default paths that was created using Cloud Console or `gcloud`...?

Reply:
Commenting on this would add clarity, but giving a stronger guarantee than GCP does might be unwise, as this scheme might change in future (and we might still read `creds.JSON`).

One path I anticipate is GCP switching the scheme (today ECDSA is popular for this), implementing key rotation (ECDSA are to be retired every so many signings), and using something the likes of the upcoming *runtime config* to distribute them.

Reply:
`lazyCredsOpener` is supposed to set any ‘Application Default Credentials’, therefore I've skipped this comment altogether.